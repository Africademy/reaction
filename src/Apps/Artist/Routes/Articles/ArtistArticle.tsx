import { Sans, Serif } from "@artsy/palette"
import React, { SFC } from "react"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Image } from "Styleguide/Elements/Image"
import { Separator } from "Styleguide/Elements/Separator"
import { Spacer } from "Styleguide/Elements/Spacer"
import { Responsive } from "Utils/Responsive"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
}

export const ArticleItem: SFC<ArticleItemProps> = props => {
  return (
    <Responsive>
      {({ xs }) => {
        if (xs) return <SmallArticleItem {...props} />
        else return <LargeArticleItem {...props} />
      }}
    </Responsive>
  )
}

const LargeArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title } = props

  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Date size="3" weight="medium">
            {date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={href} className="noUnderline">
                  {title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <Image width="135px" height="85px" src={imageUrl} />
        </Col>
      </Row>

      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      <Separator />
      <Spacer mt={3} />
    </React.Fragment>
  )
}

const SmallArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title } = props

  return (
    <React.Fragment>
      <Flex justifyContent="space-between">
        <Box pr={3}>
          <Date size="1" weight="medium">
            {date}
          </Date>
          <Spacer mb={0.5} />
          <Title size="2">
            <a href={href} className="noUnderline">
              {title}
            </a>
          </Title>
          <Spacer mb={0.5} />
          <Credit size="1" color="black60">
            {author}
          </Credit>
        </Box>
        <Image width="80px" height="50px" src={imageUrl} />
      </Flex>

      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      <Separator />
    </React.Fragment>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
