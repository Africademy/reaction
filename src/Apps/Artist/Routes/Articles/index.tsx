import { Articles_artist } from "__generated__/Articles_artist.graphql"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { ArticlesRefetchContainer as Articles } from "./ArticlesRefetchContainer"

export interface ArticlesRouteProps {
  artist: Articles_artist
}

export const ArticlesRoute: React.SFC<ArticlesRouteProps> = props => {
  return <Articles artist={props.artist as any} />
}

export const ArticlesRouteFragmentContainer = createFragmentContainer(
  ArticlesRoute,
  graphql`
    fragment Articles_artist on Artist {
      ...ArticlesRefetchContainer_artist
    }
  `
)
