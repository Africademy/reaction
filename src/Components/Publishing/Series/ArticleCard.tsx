import React, { Component  } from "react"
import styled, { StyledFunction } from "styled-components"
import { crop } from "../../../Utils/resizer"
import { pMedia } from "../../Helpers"
import { Byline } from '../Byline/Byline'
import { Fonts } from "../Fonts"
import { IconPlayCaret } from '../Icon/IconPlayCaret'

interface Props {
  article?: any,
  color?: string, 
  series?: any
}

export class ArticleCard extends Component<Props, null> {
  public static defaultProps: Partial<Props>

  render () {
    const { article, color, series } = this.props
    const { media } = article

    return (
      <ArticleCardContainer href={article.slug} color={color} className="ArticleCard">

        <TextContainer>
          <div>
            <Header>
              <div>{series.title}</div>
              {media &&
                <div>{media.duration}</div>
              }
            </Header>
            <Title>{article.title}</Title>
            <Description>{article.description}</Description>
          </div>
          <Byline
            article={article}
            color={color}
            layout='condensed'
          />
        </TextContainer>

        <ImageContainer>
          <Image src={crop(article.thumbnail_image, { width: 680, height: 450 })} />
          {media &&
            <VideoPlay><IconPlayCaret color={color} /></VideoPlay>
          }
        </ImageContainer>

      </ArticleCardContainer>
    )
  }
}

ArticleCard.defaultProps = {
  color: 'black'
}

const A: StyledFunction<Props & React.HTMLProps<HTMLLinkElement>> = styled.a
const Div: StyledFunction<Props & React.HTMLProps<HTMLDivElement>> = styled.div

const ArticleCardContainer = A`
  display: block;
  border: 1px solid;
  color: ${props => props.color};
  text-decoration: none;
  padding: 30px;
  display: flex;
  ${props => pMedia.md`
    flex-direction: column-reverse;
    padding: 20px;
  `}
`

const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
  .author, .date {
    ${Fonts.unica("s16", "medium")}
  }
  ${props => pMedia.md`
    width: 100%;
    margin-bottom: 0;
  `}
`

const Title = styled.div`
  ${Fonts.unica("s45")}
  margin-bottom: 30px;
  ${props => pMedia.md`
    ${Fonts.unica("s32")}
  `}
`

const Header = styled.div`
  ${Fonts.unica("s16", "medium")}
  display: flex;
  margin-bottom: 10px;
  div {
    margin-right: 20px;
  }
`
const Description = styled.div`
  ${Fonts.garamond("s23")}
  ${props => pMedia.md`
    ${Fonts.garamond("s19")}
    margin-bottom: 20px;
  `}
`

const ImageContainer = styled.div`
  position: relative;
  width: 50%;
  height: 100%;
  margin-left: 30px;
  ${props => pMedia.md`
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  `}
`

const Image = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
`

const VideoPlay = Div`
  position: absolute;
  left: 20px; 
  bottom: 20px;
  width: 40px;
  ${props => pMedia.md`
    width: 30px;
  `}
`
