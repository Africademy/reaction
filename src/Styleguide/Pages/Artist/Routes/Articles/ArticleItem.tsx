import { Sans, Serif } from "@artsy/palette"
import React from "react"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Image } from "Styleguide/Elements/Image"
import { Responsive } from "Styleguide/Utils/Responsive"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
}

export const ArticleItem = (props: ArticleItemProps) => {
  return (
    <Responsive>
      {({ xs }) => {
        if (xs) return <SmallArticleItem {...props} />
        else return <LargeArticleItem {...props} />
      }}
    </Responsive>
  )
}

const LargeArticleItem = (props: ArticleItemProps) => {
  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Date size="3" weight="medium">
            {props.date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={props.href} className="noUnderline">
                  {props.title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {props.author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <Image width="135px" height="85px" src={props.imageUrl} />
        </Col>
      </Row>
    </React.Fragment>
  )
}

const SmallArticleItem = (props: ArticleItemProps) => {
  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Box mb={0.5}>
            <Date size="1" weight="medium">
              {props.date}
            </Date>
          </Box>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={3}>
              <Title size="2">
                <a href={props.href} className="noUnderline">
                  {props.title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {props.author}
              </Credit>
            </Box>
            <Image width="80px" height="50px" src={props.imageUrl} />
          </Flex>
        </Col>
      </Row>
    </React.Fragment>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
