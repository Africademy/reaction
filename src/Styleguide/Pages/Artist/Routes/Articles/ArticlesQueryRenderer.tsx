import { ContextConsumer, ContextProps } from "Components/Artsy"
import React from "react"
import { graphql, QueryRenderer } from "react-relay"
import { ArticlesRefetchContainer } from "./ArticlesRefetchContainer"

interface Props extends ContextProps {
  artistID: string
}

const PAGE_SIZE = 10

export const ArticlesQueryRenderer = ContextConsumer(
  class extends React.Component<Props> {
    render() {
      const { artistID, relayEnvironment } = this.props
      return (
        <QueryRenderer
          environment={relayEnvironment}
          query={graphql`
            query ArticlesQueryRendererQuery($artistID: String!, $first: Int!) {
              artist(id: $artistID) {
                ...ArticlesRefetchContainer_artist @arguments(first: $first)
              }
            }
          `}
          variables={{ artistID, first: PAGE_SIZE }}
          render={({ props }) => {
            if (props) {
              return <ArticlesRefetchContainer artist={props.artist} />
            } else {
              return null
            }
          }}
        />
      )
    }
  }
)
