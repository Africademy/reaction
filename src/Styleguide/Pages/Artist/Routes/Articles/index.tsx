import React from "react"
import { ArticlesRefetchContainer } from "./ArticlesRefetchContainer"

interface Props {
  artist: any
}

export class ArticlesRoute extends React.Component<Props> {
  render() {
    return <ArticlesRefetchContainer artist={this.props.artist} />
  }
}
