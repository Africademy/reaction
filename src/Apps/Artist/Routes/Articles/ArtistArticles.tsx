import { ArtistArticles_artist } from "__generated__/ArtistArticles_artist.graphql"
import React, { Component } from "react"
import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
import { PaginationFragmentContainer as Pagination } from "Styleguide/Components/Pagination"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { ArticleItem } from "./ArtistArticle"

import {
  LoadingArea,
  LoadingAreaState,
} from "Apps/Artist/Components/LoadingArea"

const PAGE_SIZE = 10

interface ArtistArticlesProps {
  relay: RelayRefetchProp
  artist: ArtistArticles_artist
}

export class ArtistArticles extends Component<
  ArtistArticlesProps,
  LoadingAreaState
> {
  state = {
    isLoading: false,
  }

  loadNext = () => {
    const {
      artist: {
        articlesConnection: {
          pageInfo: { hasNextPage, endCursor },
        },
      },
    } = this.props

    if (hasNextPage) {
      this.loadAfter(endCursor)
    }
  }

  loadAfter = cursor => {
    this.toggleLoading(true)

    this.props.relay.refetch(
      {
        first: PAGE_SIZE,
        after: cursor,
        artistID: this.props.artist.id,
        before: null,
        last: null,
      },
      null,
      error => {
        this.toggleLoading(false)

        if (error) {
          console.error(error)
        }
      }
    )
  }

  toggleLoading = isLoading => {
    this.setState({
      isLoading,
    })
  }

  render() {
    return (
      <React.Fragment>
        <span id="jumpto--artistArticles" />

        <Row>
          <Col>
            <LoadingArea isLoading={this.state.isLoading}>
              {this.props.artist.articlesConnection.edges.map(
                ({ node }, index) => {
                  return (
                    <ArticleItem
                      title={node.thumbnail_title}
                      imageUrl={node.thumbnail_image.resized.url}
                      date={node.published_at}
                      author={node.author.name}
                      href={node.href}
                      key={index}
                    />
                  )
                }
              )}
            </LoadingArea>
          </Col>
        </Row>
        <Row>
          <Col>
            <Flex mb={2} justifyContent="flex-end">
              <Pagination
                pageCursors={
                  this.props.artist.articlesConnection.pageCursors as any
                }
                onClick={this.loadAfter}
                onNext={this.loadNext}
                scrollTo="#jumpto-RouteTabs"
              />
            </Flex>
          </Col>
        </Row>
      </React.Fragment>
    )
  }
}

export const ArtistArticlesRefetchContainer = createRefetchContainer(
  ArtistArticles,
  {
    artist: graphql`
      fragment ArtistArticles_artist on Artist
        @argumentDefinitions(
          first: { type: "Int", defaultValue: 10 }
          last: { type: "Int" }
          after: { type: "String" }
          before: { type: "String" }
        ) {
        id
        articlesConnection(
          first: $first
          after: $after
          before: $before
          last: $last
          sort: PUBLISHED_AT_DESC
          in_editorial_feed: true
        ) {
          pageInfo {
            hasNextPage
            endCursor
          }
          pageCursors {
            ...Pagination_pageCursors
          }
          edges {
            node {
              href
              thumbnail_title
              author {
                name
              }
              published_at(format: "MMM d, YYYY")
              thumbnail_image {
                resized(width: 300) {
                  url
                }
              }
              href
            }
          }
        }
      }
    `,
  },
  graphql`
    query ArtistArticlesQuery(
      $first: Int
      $last: Int
      $after: String
      $before: String
      $artistID: String!
    ) {
      artist(id: $artistID) {
        ...ArtistArticles_artist
          @arguments(first: $first, last: $last, after: $after, before: $before)
      }
    }
  `
)
