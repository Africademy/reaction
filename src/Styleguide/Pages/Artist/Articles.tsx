import React from "react"
import {
  graphql,
  QueryRenderer,
  createRefetchContainer,
  RelayRefetchProp,
} from "react-relay"
import { ContextProps, ContextConsumer } from "Components/Artsy"
import styled from "styled-components"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Separator } from "Styleguide/Elements/Separator"
import { ArticleItem } from "./ArticleItem"
import { Pagination } from "Styleguide/Components/Pagination"

import { Articles_artist } from "__generated__/Articles_artist.graphql"

interface ArticlesProps {
  relay: RelayRefetchProp
  artist: Articles_artist
}

const PAGE_SIZE = 10

// Renders list of articles, we'll use this in our refetch container.
class ArticlesContainer extends React.Component<ArticlesProps> {
  loadPrev = () => {
    const {
      startCursor,
      hasPreviousPage,
    } = this.props.artist.articlesConnection.pageInfo
    if (hasPreviousPage) {
      this.loadBefore(startCursor)
    }
  }

  loadNext = () => {
    const {
      hasNextPage,
      endCursor,
    } = this.props.artist.articlesConnection.pageInfo
    if (hasNextPage) {
      this.loadAfter(endCursor)
    }
  }

  loadBefore(cursor) {
    this.props.relay.refetch(
      {
        first: null,
        before: cursor,
        artistID: this.props.artist.id,
        after: null,
        last: PAGE_SIZE,
      },
      null,
      error => {
        if (error) {
          console.error(error)
        }
      }
    )
  }

  loadAfter = cursor => {
    this.props.relay.refetch(
      {
        first: PAGE_SIZE,
        after: cursor,
        artistID: this.props.artist.id,
        before: null,
        last: null,
      },
      null,
      error => {
        if (error) {
          console.error(error)
        }
      }
    )
  }

  renderPagination() {
    return (
      <div>
        <Pagination
          {...this.props.artist.articlesConnection.pageCursors}
          onClick={this.loadAfter}
          onNext={this.loadNext}
          onPrev={this.loadPrev}
        />
      </div>
    )
  }

  render() {
    return (
      <div>
        <Row>
          <Col>
            <ArticleList>
              {this.props.artist.articlesConnection.edges.map(({ node }) => {
                return (
                  <ArticleItem
                    title={node.thumbnail_title}
                    imageUrl={node.thumbnail_image.resized.url}
                    date={node.published_at}
                    author={node.author.name}
                    href={node.href}
                  />
                )
              })}
            </ArticleList>
          </Col>
        </Row>

        <Box my={2}>
          <Separator />
        </Box>

        <Row>
          <Col>
            <Flex mb={2} justifyContent="flex-end">
              {this.renderPagination()}
            </Flex>
          </Col>
        </Row>
      </div>
    )
  }
}

// This is the actual Refetch Container we want to use.
const ArticlesRefetchContainer = createRefetchContainer(
  ArticlesContainer,
  {
    artist: graphql`
      fragment Articles_artist on Artist
        @argumentDefinitions(
          first: { type: "Int" }
          last: { type: "Int" }
          after: { type: "String" }
          before: { type: "String" }
        ) {
        id
        articlesConnection(
          first: $first
          after: $after
          before: $before
          last: $last
          sort: PUBLISHED_AT_DESC
          in_editorial_feed: true
        ) {
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
          pageCursors {
            around {
              cursor
              page
              isCurrent
            }
            first {
              cursor
              page
              isCurrent
            }
            last {
              cursor
              page
              isCurrent
            }
          }
          edges {
            node {
              href
              thumbnail_title
              author {
                name
              }
              published_at(format: "MMM d, YYYY")
              thumbnail_image {
                resized(width: 300) {
                  url
                }
              }
              href
            }
          }
        }
      }
    `,
  },
  graphql`
    query ArticlesQuery(
      $first: Int
      $last: Int
      $after: String
      $before: String
      $artistID: String!
    ) {
      artist(id: $artistID) {
        ...Articles_artist
          @arguments(first: $first, last: $last, after: $after, before: $before)
      }
    }
  `
)

interface Props extends ContextProps {
  artistID: string
}

class ArticlesQueryRenderer extends React.Component<Props> {
  render() {
    const { artistID, relayEnvironment } = this.props
    return (
      <QueryRenderer
        environment={relayEnvironment}
        query={graphql`
          query ArticlesIndexQuery($artistID: String!, $first: Int!) {
            artist(id: $artistID) {
              ...Articles_artist @arguments(first: $first)
            }
          }
        `}
        variables={{ artistID, first: PAGE_SIZE }}
        render={({ props }) => {
          if (props) {
            return <ArticlesRefetchContainer artist={props.artist} />
          } else {
            return null
          }
        }}
      />
    )
  }
}

export const ArticlesContent = ContextConsumer(ArticlesQueryRenderer)

const ArticleList = styled.div``
