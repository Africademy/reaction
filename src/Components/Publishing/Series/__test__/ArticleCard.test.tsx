import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { SeriesArticle, StandardArticle, VideoArticle } from "../../Fixtures/Articles"
import { ArticleCard } from "../ArticleCard"

it("renders an article properly", () => {
  const articleCard = renderer.create(<ArticleCard article={StandardArticle} series={SeriesArticle} />).toJSON()
  expect(articleCard).toMatchSnapshot()
})

it("renders a video article properly", () => {
  const articleCard = renderer.create(<ArticleCard article={StandardArticle} series={VideoArticle} />).toJSON()
  expect(articleCard).toMatchSnapshot()
})
