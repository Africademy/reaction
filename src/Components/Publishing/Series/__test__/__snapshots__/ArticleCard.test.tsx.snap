// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a video article properly 1`] = `
.c6 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  margin: 10px 30px 0 0;
}

.c6.date {
  white-space: nowrap;
}

.c0 {
  display: block;
  border: 1px solid;
  color: black;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c1 .author,
.c1 .date {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c3 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c2 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 10px;
}

.c2 div {
  margin-right: 20px;
}

.c4 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
}

.c5 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 span {
  margin-right: 30px;
}

.c7 {
  position: relative;
  width: 50%;
  height: 100%;
  margin-left: 30px;
}

.c8 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
}

.c9 {
  position: absolute;
  left: 20px;
  bottom: 20px;
  width: 40px;
}

@media (max-width:720px) {
  .c6 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    margin: 10px 20px 0 0;
  }
}

@media (max-width:900px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c3 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c4 {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
    margin-bottom: 20px;
  }
}

@media (max-width:900px) {
  .c7 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

@media (max-width:900px) {
  .c9 {
    width: 30px;
  }
}

<a
  className="ArticleCard c0"
  color="black"
  href="joanne-artman-gallery-poetry-naturerefinement-form"
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2"
      >
        <div>
          The Future of Art
        </div>
        <div>
          03:12
        </div>
      </div>
      <div
        className="c3"
      >
        New Study Shows the Gender Pay Gap for Artists Is Not So Simple
      </div>
      <div
        className="c4"
      >
        The elegant spiral of the Nautilus shell, the sinuous pattern of the banks of a river, or the swirling vortex street of clouds - patterns exist.
      </div>
    </div>
    <div
      className="c5"
    >
      <span>
        Coming Soon
      </span>
      <div
        className="date c6"
      >
        Aug 28, 2018
      </div>
    </div>
  </div>
  <div
    className="c7"
  >
    <img
      className="c8"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FwHFgQlrTrHav5O6bQRJ0dg%252FUntitled%2BSuspended_30x67x33%2B%25282%2529_sm%2Bcropped.jpg&width=680&height=450&quality=95"
    />
    <div
      className="c9"
    >
      <svg
        className="IconPlayCaret"
        version="1.1"
        viewBox="0 0 40 56"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="black"
            points="0 0 0 56 39.3600006 28"
          />
        </g>
      </svg>
    </div>
  </div>
</a>
`;

exports[`renders an article properly 1`] = `
.c7 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  margin: 10px 30px 0 0;
}

.c7.date {
  white-space: nowrap;
}

.c6 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  margin: 10px 30px 0 0;
}

.c6.date {
  white-space: nowrap;
}

.c6:before {
  content: "";
  display: inline-block;
  min-width: 10px;
  min-height: 10px;
  border-radius: 50%;
  margin: 6px 10px 1px 0;
  background-color: black;
}

.c6:before {
  min-width: 8px;
  min-height: 8px;
  margin: 0 5px 1px 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: black;
}

.c0 {
  display: block;
  border: 1px solid;
  color: black;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c1 .author,
.c1 .date {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c3 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c2 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 10px;
}

.c2 div {
  margin-right: 20px;
}

.c4 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
}

.c8 {
  position: relative;
  width: 50%;
  height: 100%;
  margin-left: 30px;
}

.c9 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
}

@media (max-width:720px) {
  .c7 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    margin: 10px 20px 0 0;
  }
}

@media (max-width:720px) {
  .c6 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    margin: 10px 20px 0 0;
  }
}

@media (max-width:720px) {
  .c6:before {
    min-width: 8px;
    min-height: 8px;
  }
}

@media (max-width:900px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c3 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c4 {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
    margin-bottom: 20px;
  }
}

@media (max-width:900px) {
  .c8 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

<a
  className="ArticleCard c0"
  color="black"
  href="new-yorks-next-art-district"
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2"
      >
        <div>
          The Future of Art
        </div>
      </div>
      <div
        className="c3"
      >
        New York's Next Art District
      </div>
      <div
        className="c4"
      >
        Land exhibitions, make influential contacts, and gain valuable feedback about your work.
      </div>
    </div>
    <div
      className="Byline c5"
      color="black"
    >
      <div
        className="author c6"
        color="black"
      >
        By 
        Casey Lesser
      </div>
      <div
        className="date c7"
      >
        May 19, 2017
      </div>
    </div>
  </div>
  <div
    className="c8"
  >
    <img
      className="c9"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F7lsxxsw0qPAuKl37jEYitw%252Farticle%2Basset%2B1-hig%2Bres%2Bcopy.jpg&width=680&height=450&quality=95"
    />
  </div>
</a>
`;
